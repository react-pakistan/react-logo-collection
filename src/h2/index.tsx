import styled, { ThemedStyledProps } from 'styled-components';
import { ITheme } from '../theme';

export const H2 = styled.h2<IH2Props>`
  margin: 0;
  padding: 0;
  font-style: normal;
  font-stretch: normal;
  font-family: ${(props : ThemedStyledProps<IH2Props, ITheme>) : string =>
    props.fontFamily || props.theme.fontFamilyH2};
  font-size: ${(props : ThemedStyledProps<IH2Props, ITheme>) : string =>
    props.fontSize || props.theme.fontSizeH2};
  font-weight: ${(props : ThemedStyledProps<IH2Props, ITheme>) : number =>
    props.fontWeight || props.theme.fontWeightH2};
  line-height: ${(props : ThemedStyledProps<IH2Props, ITheme>) : string =>
    props.lineHeight || props.theme.lineHeightH2};
  letter-spacing: ${(props : ThemedStyledProps<IH2Props, ITheme>) : string =>
    props.letterSpacing || props.theme.letterSpacingH2};
  text-align: ${(props : ThemedStyledProps<IH2Props, ITheme>) : string =>
    props.textAlign || props.theme.textAlignH2};
  text-transform: ${(props : ThemedStyledProps<IH2Props, ITheme>) : string =>
    props.textTransform || props.theme.textTransformH2};
  color: ${(props : ThemedStyledProps<IH2Props, ITheme>) : string =>
    props.color || props.theme.colorH2}
`;

export interface IH2Props {
  /**
   * Font Family property for H1 tag
   * @default Montserrat
   */
  fontFamily : string;
  /**
   * Font Size property for H1 tag
   * @default ThemeH2
   */
  fontSize? : string;
  /**
   * Font Weight property for H1 tag
   * @default ThemeH2
   */
  fontWeight? : number;
  /**
   * Line Height proprty for H1 tag
   * @default ThemeH2
   */
  lineHeight? : string;
  /**
   * Letter Spacing property for H1 tag
   * @default ThemeH2
   */
  letterSpacing? : string;
  /**
   * Text Align property for H1 tag
   * @default ThemeH2
   */
  textAlign? : string;
  /**
   * Text Transform property for H1 tag
   * @default ThemeH2
   */
  textTransform? : string;
  /**
   * Color property for H1 tag
   * @default ThemeH2
   */
  color? : string;
}
