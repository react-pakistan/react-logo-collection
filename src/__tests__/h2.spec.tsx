import React from 'react';
import { shallow } from 'enzyme';
import { H2 } from '../h2';
import { h2Props } from '../__stories__/components/h2.story';

describe('Testing H2 component', () => {
  it('matches H2 snapshot', () => {
    const wrapper = shallow(<H2 {...h2Props()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
