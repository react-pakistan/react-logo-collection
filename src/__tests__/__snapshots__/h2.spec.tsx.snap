// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing H2 component matches H2 snapshot 1`] = `ShallowWrapper {}`;
